api/proto/core.proto

```
@@ -0,0 +1,93 @@
+  
+  // GetRoutes returns all configured routes
+  rpc GetRoutes(GetRoutesRequest) returns (GetRoutesResponse);
+  
+  // GetExternalRoutes returns only external (non-container) routes
+  rpc GetExternalRoutes(GetExternalRoutesRequest) returns (GetExternalRoutesResponse);
+  
+  // NotifyImagePushed is called by the registry when an image is pushed
+  rpc NotifyImagePushed(NotifyImagePushedRequest) returns (NotifyImagePushedResponse);
+  
+  // WatchRouteChanges streams route change events for cache invalidation
+  rpc WatchRouteChanges(WatchRouteChangesRequest) returns (stream RouteChangeEvent);
```

Buf (1.64.0)

[error] RPC response type "RouteChangeEvent" should be named "WatchRouteChangesResponse" or "CoreServiceWatchRouteChangesResponse".

(RPC_RESPONSE_STANDARD_NAME)